pallets/api/src/messaging/mod.rs

```diff
@@ -435,67 +478,40 @@ pub mod pallet {
 			Ok(())
 		}
 
-		// Remove a request/response, returning any deposit previously taken.
+		/// Try and remove a collection of messages.
+		/// Will revert if any one of the messages is erroneous.
+		#[frame_support::transactional]
 		#[pallet::call_index(5)]
-		#[pallet::weight(Weight::zero())] // todo: benchmarking after consolidating storage
+		#[pallet::weight(Weight::zero())]
 		pub fn remove(
 			origin: OriginFor<T>,
 			messages: BoundedVec<MessageId, T::MaxRemovals>,
 		) -> DispatchResult {
 			let origin = ensure_signed(origin)?;
+
```

Comment:
likely out of scope for this PR, but right now anyone can remove a message. Should there be limited origins for this?

Reply:
i cant think of what other origin would be useful for now.

Reply:
Shouldn't only the creator of the message be able to remove it?